internal/juju/client.go

```diff
@@ -83,8 +86,51 @@ func (cf *ConnectionFactory) GetConnection(model *string) (api.Connection, error
 
 	conn, err := connr.Connect()
 	if err != nil {
-		log.Error().Err(err).Msg("connection not established")
+		cf.Errorf(err, "connection not established")
 		return nil, err
 	}
 	return conn, nil
 }
+
+// module names for logging
+// @module=juju.<subsystem>
+// e.g.:
+//
+//	@module=juju.client
+const LogJujuClient = "client"
+
+// TODO hml 04-Aug-2023
+// Investigate if the context from terraform can be passed in
+// and used with tflog here, for now, use context.Background.
+
+func (cf *ConnectionFactory) Debugf(msg string, additionalFields ...map[string]interface{}) {
+	if cf.subCtx == nil {
+		cf.subCtx = tflog.NewSubsystem(context.Background(), LogJujuClient)
```

Comment:
Should we use `context.TODO` here, as it's  not super clear which context it should be? might be more appropriate, though `Background` makes sense too, I'm not entirely sure, just wanted to point out

Reply:
It's highly unlikely we'll get a context here, nor do I believe it's useful to pipe one in from the provider code just for logging. However a non nil context is required.